WordPress/openverse
# Add new SensitiveMedia and MediaReport models

## Fixes



Fixes #2626 by @AetherUnbound 

## Description


This PR adds the models as described in the [Steps 1 and 2 of the Implementation Plan/ API](https://docs.openverse.org/projects/proposals/trust_and_safety/content_report_moderation/20230517-implementation_plan_mature_copy_change.html#api)

@WordPress/openverse-api , I need help figuring out why these changes cause one sensitive table db query per result, both for search and for the single result requests:
<details>
<summary>Sample search query</summary>
```
E           1. SELECT "image"."id", "image"."created_on", "image"."updated_on", "image"."identifier", "image"."foreign_identifier", "image"."title", "image"."foreign_landing_url", "image"."creator", "image"."creator_url", "image"."thumbnail", "image"."provider", "image"."url", "image"."filesize", "image"."filetype", "image"."watermarked", "image"."license", "image"."license_version", "image"."source", "image"."last_synced_with_source", "image"."removed_from_source", "image"."view_count", "image"."tags", "image"."category", "image"."meta_data", "image"."width", "image"."height", "api_matureimage"."created_on", "api_matureimage"."identifier" FROM "image" LEFT OUTER JOIN "api_matureimage" ON ("image"."identifier" = "api_matureimage"."identifier") WHERE (NOT ("image"."provider" IN (SELECT U0."provider_identifier" FROM "content_provider" U0 WHERE U0."filter_content") AND "image"."provider" IS NOT NULL) AND "image"."identifier" IN ('6b7d6870ab734fe7b187d5509f408b48'::uuid, 'f890597b52c8432fbd86a8f4362182af'::uuid, '5aa22382cc77481eafb88693b304a44e'::uuid, '676777e3bb9c432eb8c80ce27f6a639b'::uuid, '4657177f42764b379e9cfc2e88b6b996'::uuid, '32a1ccf579d84eafb974045d93ea769e'::uuid, '69a7b7fda0c9497e8ec7f3a2177298fd'::uuid, 'caefbd828ae145e2b5f5d901778198f4'::uuid, '06f5b2e2f2c546e789e5d7c2fe2330f6'::uuid, '467d9209e90a4f469447d4511ef00bdf'::uuid, '475ead8a9fa4481c999c101b8407bd30'::uuid, 'cb64afe904864da8a5c46714bf72b02d'::uuid, '9c6571bf310740b6baab3a7e170ae301'::uuid, '6af6e4ca356c4c828047a3d0fbede4bb'::uuid, '8ca3fb5ed56747c9abc7e9c4b7374c92'::uuid, 'c2020b32f0c04b20a714a74a3750d8e2'::uuid, 'acabd6bf5a774289a587918dac0eb66b'::uuid, 'ea4edd319f974cdaa02a43d85bce6674'::uuid, '0f5bf191878b45588674f22a210a80b0'::uuid, '798eea993dd74aa5a9a92acfd281a573'::uuid)) ORDER BY "image"."created_on" DESC
E           2. SELECT "api_sensitiveimage"."created_on", "api_sensitiveimage"."identifier" FROM "api_sensitiveimage" WHERE "api_sensitiveimage"."identifier" = '6b7d6870ab734fe7b187d5509f408b48'::uuid LIMIT 21
```
</details>

## Testing Instructions

1. Checkout `main` branch and mark several items as`mature` in the admin (http://localhost:50280/admin, user `deploy`, password `deploy`). I used a search for "Couple Food Photo" and marked a couple of results as sensitive.
2. Checkout this branch and run `just api/dj migrate`. This should create the new db tables.
3. Go to admin and check that you can still see the old reports.
4. Create some new reports.

## Checklist



[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
